eclipse/src/saros/Saros.java

```diff
@@ -130,10 +135,17 @@ public void start(BundleContext context) throws Exception {
 
     isLifecycleStarted = true;
 
-    getWorkbench().addWorkbenchListener(workbenchShutdownListener);
     isInitialized = true;
   }
 
+  @Inject
+  @Optional
+  public void applicationStarted(
+      @UIEventTopic(UIEvents.UILifeCycle.APP_STARTUP_COMPLETE) Event event, Saros saros) {
+
+    saros.getWorkbench().addWorkbenchListener(saros.workbenchShutdownListener);
```

Comment:
I do not understand the benefit ?

Reply:
This Listener was previously added in Line 133.
But with the Addition of Eclipse 4 Fragments Saros is initialized before the Workbench and the Listener has to be added seperately after the Startup is complete and the Workbench exists.

Reply:
The listener is still removed in the stop method. This seems pretty confusing. Isn't there no new method that should be added ?

Reply:
For us it seemed that the addition of E4 components moved the startup of saros up, before the initialization of a workbench.
Thats why we moved the addition of the workbench listener it to this extra function.
But the time whene the stop method is called didnt change so there was no need to move the remove call to an extra function.
We realized that there is some redundancy though, since the workbench listener was and still is called before the stop method, which both call stopLifeCycle() and we saw no change when removing the listener completely.
But we guessed that the listener serves a purpose, probably when the workbench isnt shutdown by ordinary means which would not stop all plugins

Reply:
https://github.com/saros-project/saros/commit/d602c26eef4185307d05270a803c7333357d9eab#diff-677f70b4646a5136f14fdb947bca53f9afedec5f58e09d5f9f201315fca8e4c3

Reply:
I subscribe to UIEvents.UILifeCycle.APP_SHUTDOWN_STARTED which seems to be the counterpart. The only issue I found that it was called twice.